### PR TITLE
[JIT] remove specialized list ops

### PR DESCRIPTION
**Pull request by eellison**
Now that lists are no longer specialized, we can register only one operator for list ops that are generic to their element type.
This PR reorgs lists into three sets of ops:
- CREATE_GENERIC_LIST_OPS
- CREATE_SPECIALIZED_LIST_OPS
- CREATE_COMPARATOR_LIST_OPS_SPECIALIZED (we didn't bind certain specialized ops to Tensor)

This is important to land quickly because mobile is finalizing its bytecode soon, after which we could not remove these ops.

### CODE REVIEW

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 1]
```diff
@@ -2300,173 +2301,91 @@ RegisterOperators reg2({
...
-        listAppend<std::string>,
-        aliasAnalysisFromSchema()),
+        " decl_type " el) -> int",                         \
+          listCount<value_type>,                           \
+          aliasAnalysisFromSchema()),
 
-#undef CREATE_IMMUTABLE_LIST_OPS
-#undef CREATE_MUTABLE_LIST_OPS
+    CREATE_SPECIALIZED_LIST_OPS("int", int64_t)
+        CREATE_SPECIALIZED_LIST_OPS("float", double)
```
Comment (iseeyuan) [inline torch/csrc/jit/runtime/register_prim_ops.cpp]: Indent?

**Review by iseeyuan — APPROVED** [review]
LGTM. Thanks for the quick fix!

File: torch/csrc/jit/runtime/register_prim_ops.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/runtime/register_prim_ops.cpp]: this is auto-formatting :'0

**Comment by iseeyuan** [comment]
Please import to Phabricator so that we could monitor the effects on the internal models.

**Review by iseeyuan — APPROVED** [review]

**Comment by eellison** [comment]
Ninja fix is in for master breakage

**Pull request closed without merging**